java/arcs/core/data/expression/PaxelParser.kt

```diff
@@ -130,7 +130,7 @@ object PaxelParser {
             )
         }
 
-    private val scopeQualifier = functionCall / ident.map { FieldExpression<Any>(null, it, false) }
+    private val scopeQualifier = functionCall / ident.map { FieldExpression<Any>(null, it, false) } / parser(::nestedExpression)
 
```

Comment:
the problem is, PrimaryExpression looks like this:

```
PrimaryExpression ::= '(' nestedExpression ')'
| ...
| scopeLookup
```

Since the prefix of scopeLookup is also nested expression, the first match wins and parses as a nested expression instead of the beginning of a scopeLookup. I'll try reordering primary expression to see if it fixes it.

Reply:
Ok, the fix is just to remove 'nestedExpression' from primaryExpression. Since

```
scopeLookoup ::= scopeQualifier ('.' ident)*
```

and 
```
scopeQualifier = ... / nestedExpression
```

Then primary expression also includes nested expression when no trailing '.' occur.